Importer Rework - ComplianceNetworkZone does not contain a definition for CalculateInternetZone

In v9.0 upgraded customer environment N001 test stage, we get the following unit test error 

error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateInternetZone' and no accessible extension method 'CalculateInternetZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) 

Tested with Laennart:feat/cisco-asa-import (which is is sync with cactus:importer-rework)

@Robin-Smets could you pls have a look at this error?


```
TASK [tests-unit : run csharp tests] ***************************************************************************************************************************************************************

fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["dotnet", "test"], "delta": "0:00:09.401619", "end": "2025-10-19 10:11:51.712381", "msg": "non-zero return code", "rc": 1, "start": "2025-10-19 10:11:42.310762", "stderr": "", "stderr_lines": [], "stdout": "  Determining projects to restore...\n  Restored /usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj (in 518 ms).\n  19 of 20 projects are up-to-date for restore.\n  FWO.Basics -> /usr/local/fworch/lib/files/FWO.Basics/bin/Debug/net8.0/FWO.Basics.dll\n  FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Debug/net8.0/FWO.Logging.dll\n  FWO.Encryption -> /usr/local/fworch/lib/files/FWO.Encryption/bin/Debug/net8.0/FWO.Encryption.dll\n  FWO.Data -> /usr/local/fworch/lib/files/FWO.Data/bin/Debug/net8.0/FWO.Data.dll\n  FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Debug/net8.0/FWO.Config.File.dll\n  FWO.Mail -> /usr/local/fworch/lib/files/FWO.Mail/bin/Debug/net8.0/FWO.Mail.dll\n  FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Debug/net8.0/FWO.Api.Client.dll\n  FWO.Middleware.Client -> /usr/local/fworch/lib/files/FWO.Middleware.Client/bin/Debug/net8.0/FWO.Middleware.Client.dll\n  FWO.Recert -> /usr/local/fworch/lib/files/FWO.Recert/bin/Debug/net8.0/FWO.Recert.dll\n  FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Debug/net8.0/FWO.Config.Api.dll\n  FWO.ExternalSystems -> /usr/local/fworch/lib/files/FWO.ExternalSystems/bin/Debug/net8.0/FWO.ExternalSystems.dll\n  FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Debug/net8.0/FWO.Report.Filter.dll\n  FWO.FwLogic -> /usr/local/fworch/lib/files/FWO.FwLogic/bin/Debug/net8.0/FWO.FwLogic.dll\n  FWO.DeviceAutoDiscovery -> /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/bin/Debug/net8.0/FWO.DeviceAutoDiscovery.dll\n  FWO.Services -> /usr/local/fworch/lib/files/FWO.Services/bin/Debug/net8.0/FWO.Services.dll\n  FWO.Report -> /usr/local/fworch/lib/files/FWO.Report/bin/Debug/net8.0/FWO.Report.dll\n  FWO.Compliance -> /usr/local/fworch/lib/files/FWO.Compliance/bin/Debug/net8.0/FWO.Compliance.dll\n  FWO.Ui -> /usr/local/fworch/ui/files/FWO.UI/bin/Debug/net8.0/FWO.Ui.dll\n  FWO.Middleware.Server -> /usr/local/fworch/middleware/files/FWO.Middleware.Server/bin/Debug/net8.0/FWO.Middleware.Server.dll\n/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(55,26): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateInternetZone' and no accessible extension method 'CalculateInternetZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]\n/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(116,26): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateInternetZone' and no accessible extension method 'CalculateInternetZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]\n/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(158,23): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateLocalZone' and no accessible extension method 'CalculateLocalZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]\n/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(210,23): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateLocalZone' and no accessible extension method 'CalculateLocalZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]", "stdout_lines": ["  Determining projects to restore...", "  Restored /usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj (in 518 ms).", "  19 of 20 projects are up-to-date for restore.", "  FWO.Basics -> /usr/local/fworch/lib/files/FWO.Basics/bin/Debug/net8.0/FWO.Basics.dll", "  FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Debug/net8.0/FWO.Logging.dll", "  FWO.Encryption -> /usr/local/fworch/lib/files/FWO.Encryption/bin/Debug/net8.0/FWO.Encryption.dll", "  FWO.Data -> /usr/local/fworch/lib/files/FWO.Data/bin/Debug/net8.0/FWO.Data.dll", "  FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Debug/net8.0/FWO.Config.File.dll", "  FWO.Mail -> /usr/local/fworch/lib/files/FWO.Mail/bin/Debug/net8.0/FWO.Mail.dll", "  FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Debug/net8.0/FWO.Api.Client.dll", "  FWO.Middleware.Client -> /usr/local/fworch/lib/files/FWO.Middleware.Client/bin/Debug/net8.0/FWO.Middleware.Client.dll", "  FWO.Recert -> /usr/local/fworch/lib/files/FWO.Recert/bin/Debug/net8.0/FWO.Recert.dll", "  FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Debug/net8.0/FWO.Config.Api.dll", "  FWO.ExternalSystems -> /usr/local/fworch/lib/files/FWO.ExternalSystems/bin/Debug/net8.0/FWO.ExternalSystems.dll", "  FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Debug/net8.0/FWO.Report.Filter.dll", "  FWO.FwLogic -> /usr/local/fworch/lib/files/FWO.FwLogic/bin/Debug/net8.0/FWO.FwLogic.dll", "  FWO.DeviceAutoDiscovery -> /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/bin/Debug/net8.0/FWO.DeviceAutoDiscovery.dll", "  FWO.Services -> /usr/local/fworch/lib/files/FWO.Services/bin/Debug/net8.0/FWO.Services.dll", "  FWO.Report -> /usr/local/fworch/lib/files/FWO.Report/bin/Debug/net8.0/FWO.Report.dll", "  FWO.Compliance -> /usr/local/fworch/lib/files/FWO.Compliance/bin/Debug/net8.0/FWO.Compliance.dll", "  FWO.Ui -> /usr/local/fworch/ui/files/FWO.UI/bin/Debug/net8.0/FWO.Ui.dll", "  FWO.Middleware.Server -> /usr/local/fworch/middleware/files/FWO.Middleware.Server/bin/Debug/net8.0/FWO.Middleware.Server.dll", "/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(55,26): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateInternetZone' and no accessible extension method 'CalculateInternetZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]", "/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(116,26): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateInternetZone' and no accessible extension method 'CalculateInternetZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]", "/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(158,23): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateLocalZone' and no accessible extension method 'CalculateLocalZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]", "/usr/local/fworch/test/csharp/FWO.Test/SpecialZoneTest.cs(210,23): error CS1061: 'ComplianceNetworkZone' does not contain a definition for 'CalculateLocalZone' and no accessible extension method 'CalculateLocalZone' accepting a first argument of type 'ComplianceNetworkZone' could be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]"]}
```

--- a/roles/tests-unit/tasks/main.yml
+++ b/roles/tests-unit/tasks/main.yml
@@ -1,36 +1,45 @@
 - block:
-    - name: sync test files to frontend target
-      synchronize:
-        src: FWO.Test
-        dest: "{{ test_dir }}/csharp"
-        rsync_opts:
-          - "--chown={{ fworch_user }}:{{ fworch_group }}"
-          - "--delete"
-      become: true
+      - name: ensure test_dir exists
+        file:
+            path: "{{ test_dir }}/csharp"
+            state: directory
+            owner: "{{ fworch_user }}"
+            group: "{{ fworch_group }}"
+            mode: "0755"
+        become: true
 
-    - name: run csharp tests
-      command: dotnet test
-      args:
-        chdir: "{{ csharp_test_start_dir }}"
-      become: true
-      become_user: "{{ fworch_user }}"
-      register: csharp_tests
-      ignore_errors: false
-      environment: "{{ proxy_env }}"
+      - name: sync test files to frontend target
+        synchronize:
+            src: FWO.Test
+            dest: "{{ test_dir }}/csharp"
+            rsync_opts:
+                - "--chown={{ fworch_user }}:{{ fworch_group }}"
+                - "--delete"
+        become: true
 
-    - name: show csharp test results in case of errors
-      debug:
-        var: csharp_tests
-      when: csharp_tests.rc != 0
+      - name: run csharp tests
+        command: dotnet test
+        args:
+            chdir: "{{ csharp_test_start_dir }}"
+        become: true
+        become_user: "{{ fworch_user }}"
+        register: csharp_tests
+        ignore_errors: false
+        environment: "{{ proxy_env }}"
+
+      - name: show csharp test results in case of errors
+        debug:
+            var: csharp_tests
+        when: csharp_tests.rc != 0
 
   when: "'frontends' in group_names"
   # run only once (on frontends)
 
 - block:
-    - name: Run importer Python unit tests
-      environment:
-        PYTHONPATH: "/usr/local/fworch/importer"
-      args:
-        chdir: "/usr/local/fworch/importer/test"
-      command: >
-        {{ importer_venv_dir }}/bin/python3 -m unittest discover -s . -p "test_*.py" -v
+      - name: Run importer Python unit tests
+        environment:
+            PYTHONPATH: "/usr/local/fworch/importer"
+        args:
+            chdir: "/usr/local/fworch/importer/test"
+        command: >
+            {{ importer_venv_dir }}/bin/python3 -m unittest discover -s . -p "test_*.py" -v

